### PR TITLE
Simplify credential usage.

### PR DESCRIPTION
**Pull request by balamurugana**
## How Has This Been Tested?

`make test`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/credential.go [hunk 1]
```diff
@@ -75,65 +58,71 @@ type credential struct {
...
+	return isAccessKeyValid(cred.AccessKey) && isSecretKeyValid(cred.SecretKey)
 }
 
-// Initialize a new credential object
-func newCredential() credential {
-	return newCredentialWithKeys(mustGetAccessKey(), mustGetSecretKey())
-}
+// Equals - returns whether two credentials are equal or not.
+func (cred credential) Equal(ccred credential) bool {
+	if !ccred.IsValid() {
```
Comment (donatello) [inline cmd/credential.go]: This check seems unnecessary in the Equal() function, because `createCredential` is already validating.

Comment (balamurugana) [reply cmd/credential.go]: Passed `ccred` may be from empty or invalid due to JSON unmarshaling.  This check requires in this case.

Comment (krishnasrinivas) [reply cmd/credential.go]: @balamurugana can you point the code flow? because `authenticateJWT` is calling `createCredential`

Comment (balamurugana) [reply cmd/credential.go]: @krishnasrinivas I am unable to understand what code flow you are asking.

This method checks for equality of two credentials.  Those credentials could be created with different method than `createCredential()` or `mustGetNewCredential()`.  `credential.Equal()` prevents such error case by validating passed `ccred` is valid or not.

In short, if both `cred` and `ccred` are empty, this method makes equality check failure.

File: cmd/credential.go [hunk 2]
```diff
@@ -75,65 +58,71 @@ type credential struct {
...
+		return false
+	}
 
-func newCredentialWithKeys(accessKey, secretKey string) credential {
-	secretHash := mustGetHashedSecretKey(secretKey)
-	return credential{accessKey, secretKey, secretHash}
-}
+	if cred.secretKeyHash == nil {
+		secretKeyHash, err := bcrypt.GenerateFromPassword([]byte(cred.SecretKey), bcrypt.DefaultCost)
+		if err != nil {
```
Comment (harshavardhana) [inline cmd/credential.go]: Since this error is lost, it would be better we can log it.. to know why even bcrypt failed..

**Review by harshavardhana — APPROVED** [review]
One minor change..

**Comment by harshavardhana** [comment]
ping @balamurugana - can you resolve the conflicts.. ?

File: cmd/credential.go (hunk 2 above)
Comment (balamurugana) [reply cmd/credential.go]: Done

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**